src/gui/DatabaseTabWidget.cpp

```diff
@@ -336,6 +341,22 @@ bool DatabaseTabWidget::saveDatabase(Database* db, QString filePath)
         } else {
             dbStruct.modified = true;
             updateTabName(db);
+
+            if (++dbStruct.saveAttempts > 2 && useAtomicSaves) {
```

Comment:
I would check for write permissions first and show a dedicated error message. Otherwise the user will disable atomic saves for nothing if the problem is actually missing write permissions.
I know this is an edge case, because without write permissions, the DB will be opened in read-only mode, but it can still occur, if write permissions are somehow removed mid-air.

Reply:
You can't easily check for write conditions on windows. I did a fair bit of research on the matter while trying to diagnose.

Reply:
Hmm. But that would also mean that the QTemporaryFile workaround could just delete the file when it's unable to move the temporary file to the original location after deleting the old one. I don't know how it's on Windows, but when you own a file, you can always delete it, but you may not be able to create a new one in the same directory.

Reply:
I have an idea: check the return value of `tempFile.rename()` when moving the temporary file and show a save file dialog to choose a new location as long as it's returning false.

Reply:
OK, leave as is then.